var/spack/repos/builtin/packages/silo/package.py

```diff
@@ -27,15 +27,25 @@ class Silo(AutotoolsPackage):
             description='Produce position-independent code (for shared libs)')
     variant('mpi', default=True,
             description='Compile with MPI Compatibility')
+    variant('szip', default=False,
+            description='Enable szip support')
 
-    depends_on('hdf5@:1.10.999', when='@:4.10.2')
-    depends_on('hdf5~mpi', when='~mpi')
     depends_on('mpi', when='+mpi')
-    depends_on('hdf5+mpi', when='+mpi')
+    depends_on('hdf5@:1.10.999+szip', when='@:4.10.2+szip')
+    depends_on('hdf5@:1.10.999~szip', when='@:4.10.2~szip')
+    depends_on('hdf5~mpi+szip', when='~mpi+szip')
+    depends_on('hdf5+mpi+szip', when='+mpi+szip')
+    depends_on('hdf5~mpi~szip', when='~mpi~szip')
+    depends_on('hdf5+mpi~szip', when='+mpi~szip')
```

Comment:
There has been a lot of discussion about propagating variants downstream (I don't have time to find the reference) but this change needs to be reverted. Use `silo ^hdf5+szip` to demand a silo installation with szip.

Reply:
OK, I will follow your guidelines